include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -997,25 +997,27 @@ struct __parallel_find_backward_tag
 // Tag for __parallel_find_or for or-semantic
 struct __parallel_or_tag
 {
-    class __atomic_compare
+    using _AtomicType = unsigned int;
+
+    static constexpr _AtomicType __found_state = 1;
+    static constexpr _AtomicType __not_found_state = 0;
+
+    struct __compare_state
     {
-      public:
-        template <typename _LocalAtomic, typename _GlobalAtomic>
         bool
-        operator()(const _LocalAtomic& __found_local, const _GlobalAtomic& __found) const
+        operator()(const _AtomicType __found_local, const _AtomicType __found) const
```

Comment:
Am I correct that this is actually never called in the current PR?

the `__parallel_or_tag` overload takes `__comp` as a parameter but discards it without use.

Reply:
You are correct, fixed.